crates/jstzd/tests/utils.rs

```diff
@@ -17,13 +17,33 @@ pub const ACTIVATOR_SECRET_KEY: &str =
     "unencrypted:edsk31vznjHSSpGExDMHYASz45VZqXN4DPxvsa4hAyY8dHM28cZzp6";
 pub const ROLLUP_ADDRESS: &str = "sr1PuFMgaRUN12rKQ3J2ae5psNtwCxPNmGNK";
 
-pub async fn setup() -> (OctezNode, OctezClient, octez_baker::OctezBaker) {
+pub async fn poll<'a, F, T>(
```

Comment:
Nit: `retry` is effectively `poll` + predicate check. Would it be possible to implement retry using poll? `poll` might need some changes.

Reply:
Yes. I'm keeping it here as is and moving it properly in #681 since I'm adding some tests for it and thus there are quite a few lines of changes.